common/json_writer/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+use std::fmt::Write;
+
+#[derive(Debug, Clone)]
+pub struct JsonWriter {
+    buffer: String,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum JsonWriterError {
+    #[error("Failed to write the f64 value")]
+    Writef64ValueError(#[from] std::fmt::Error),
+}
+
+impl JsonWriter {
```

Comment:
I'm aware of the motivations behind this `JsonWriter ` and the complete absence of error checking. However, this is not obvious and deserves a doc comment.